### PR TITLE
CDAP-3042 fix bug where buffered operations weren't seen in multiget

### PR DESCRIPTION
**Pull request by albertshau**
Fixes a bug in BufferingTable where multiget calls for HBase were
only returning persisted data and could not see changes made in the
same transaction. In order to fix this, BufferingTable now has a
getPersisted(List<Get> gets) method for subclasses to override.
BufferingTable will then apply buffered changes on top of the
persisted changes.

Also adding a service endpoint to the UserProfiles example app to
demonstrate usage of multi-get on a Table.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-RBT336

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/BufferingTable.java [hunk 1]
```diff
@@ -406,6 +426,52 @@ public Row get(byte[] row, byte[] startColumn, byte[] stopColumn, int limit) {
...
+
+        // navigable copy of the persisted data. Implementation may return immutable or unmodifiable maps,
+        // so we make a copy here.
+        NavigableMap<byte[], byte[]> rowColumns = Maps.newTreeMap(Bytes.BYTES_COMPARATOR);
+        rowColumns.putAll(persistedRow);
+
+        byte[] row = get.getRow();
+        NavigableMap<byte[], Update> buffCols = buff.get(row);
+        // if the row was deleted in the buffer, add an empty row to the results
+        if (buffCols == null && buff.containsKey(row)) {
```
Comment (ghelmling) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/BufferingTable.java]: I see a similar check in the get(Get) method, but I don't think this is actually needed, as we never seem to put a null value for the entire row.  Instead the delete(byte[] row) method first retrieves all columns, then puts a null value for each column.

**Comment by ghelmling** [comment]
One comment, otherwise looks good.

**Comment by albertshau** [comment]
@ghelmling addressed comment, can you take another look?

**Comment by ghelmling** [comment]
lgtm

**Pull request merged**